### PR TITLE
Feature/changes v1

### PR DESCRIPTION
**Pull request by kmp24**

### CODE REVIEW

**Comment by matthewhanson** [comment]
PR for adding antimeridian handling, looping in @gadomski and @pjhartzell

File: tests/test_commands.py [hunk 1]
```diff
@@ -109,18 +111,27 @@ def test_create_item(self):
...
                         for a in d["assets"].values():
                             a["href"] = a["href"].split("data-files")[1]
 
-                        for c in d["geometry"]["coordinates"][0]:
-                            c[0] = round(c[0], 5)
-                            c[1] = round(c[1], 5)
+                        if len(d["geometry"]["coordinates"]) > 1:
+                            for i in range(0, len(d["geometry"]["coordinates"][0])):
+                                for c in d["geometry"]["coordinates"][0][i]:
+                                    c[0] = round(c[0], 5)
```
Comment (matthewhanson) [inline tests/test_commands.py:129]: In the stac.py file `bbox` and `proj_bbox` are rounded to 6 decimal places, while these are 5.
Recommend creating a constant in one of the modules for lat/lon precision and reference that where it is hardcoded.

File: src/stactools/sentinel2/commands.py [hunk 2]
```diff
@@ -55,11 +56,15 @@ def create_item_command(
             with open(providers) as f:
                 additional_providers = json.load(f)
 
+        antimeridian_strategy = "split"
+        strategy = Strategy[antimeridian_strategy.upper()]
```
Comment (pjhartzell) [inline src/stactools/sentinel2/commands.py:60]: Worth exposing the antimeridian strategy as an argument or option to the command?

File: src/stactools/sentinel2/stac.py [hunk 3]
```diff
@@ -121,15 +127,25 @@ def create_item(
...
         datetime=metadata.datetime,
-        properties={},
+        properties={"created": created},
     )
 
+    # Handle antimeridian
+    handle_am = False
+    for coord in metadata.geometry["coordinates"][0]:
+        if coord[0] >= 180 or coord[0] <= -180:
+            handle_am = True
```
Comment (pjhartzell) [inline src/stactools/sentinel2/stac.py:144]: Just clarifying, when you have a scene that crosses the antimeridian, there will be one or more points in the geometry will have a longitude value less than -180 or greater than +180? I'm accustomed to seeing a mix of positive and negative longitude values, but none with a magnitude greater than 180.

File: src/stactools/sentinel2/stac.py [hunk 4]
```diff
@@ -121,15 +127,25 @@ def create_item(
...
+
+    if handle_am is True:
+        handle_antimeridian(item, antimeridian_strategy)
```
Comment (pjhartzell) [inline src/stactools/sentinel2/stac.py:147]: The `handle_antimeridian` function was built to handle a landsat quirk where we were ingesting USGS STAC geometries that were already correctly split along the antimeridian. There is no harm in applying the same logic here, but I'm not sure if sentinel data has this same characteristic. If not, `antimeridian.fix_item` could be called directly.

**Pull request merged**